### ISSUE TITLE
Add compute function for dividing a duration by a duration

### ISSUE THREAD

**Issue by jamesdow21** [apache/arrow#37192]
### Describe the enhancement requested

Currently the supported operations between durations are addition and subtraction (both giving back a duration with the same resolution).

Division between durations also makes sense (should yield a float) but is not supported

This would also enable the equivalent of [python timedelta's `.total_seconds()` method](https://docs.python.org/3/library/datetime.html#datetime.timedelta.total_seconds), which is the same as just dividing by `timedelta(seconds=1)`

### Component(s)

C++, Python

**Comment by rok** [comment]
Thanks for reporting this!
Could you say more about your intended use of this? If you're looking to get number of units since epoch you could just run:
```python
import pyarrow as pa
arr = pa.array([0, 10E17], pa.timestamp("ns"))
arr.cast(pa.int64())
<pyarrow.lib.Int64Array object at 0x7fea336b9ea0>
[
  0,
  1000000000000000000
]
```

If you're looking to round your time to multiples of units I'd suggest looking at temporal rounding functions:
```python
import pyarrow.compute as pc
pc.floor_temporal(arr, multiple=1, unit="second")
<pyarrow.lib.TimestampArray object at 0x7fea33be2740>
[
  1970-01-01 00:00:00.000000000,
  2001-09-09 01:46:40.000000000
]
```
More here: https://arrow.apache.org/docs/cpp/compute.html#conversions

**Comment by jamesdow21** [comment]
Setup:
```
import numpy as np
import pyarrow as pa
import pyarrow.compute as pc

arr = pa.array(np.random.randint(0, 3_600_000_000, 10000, dtype="int64"), type=pa.duration("us"))
scalar = pa.scalar(1, type=pa.duration("s"))
```

As a more narrow example, what I'm currently doing (including what I believe pandas is doing under the hood) essentially works out to
```
pc.divide(
    pc.cast(pc.cast(arr, pa.int64()), pa.float64()), 
    pc.cast(pc.cast(pc.cast(scalar, arr.type), pa.int64()), pa.float64())
)
```

but it would be nice if this worked to give the same result  
`pc.divide(arr, scalar)`

**Comment by rok** [comment]
Oh I see, I think this actually should work. Is this what you'd expect?

```python
import pyarrow as pa
import pyarrow.compute as pc

arr1 = pa.array([1], type=pa.duration("s"))
arr2 = pa.array([10000], type=pa.duration("ms"))
pc.divide(arr1, arr2)
<pyarrow.lib.DoubleArray object at ...>
[
  0.1
]
```

**Comment by js8544** [comment]
This was supported recently in https://github.com/apache/arrow/pull/36800 and will be available in the next release.

**Comment by raulcd** [comment]
Just a minor nit, the issue mentioned above will be part of 14.0.0 not 13.0.0

**Comment by js8544** [comment]
> Just a minor nit, the issue mentioned above will be part of 14.0.0 not 13.0.0

Right, sorry I forgot that 13 is still not out yet. I've updated my comment above.

**Comment by jamesdow21** [comment]
> This was supported recently in #36800 and will be available in the ~next release~ 14.0.0 release.

I searched the issues to see if this was a duplicate, but forgot to search pull requests as well  

Glad to see it will be supported soon!
